pkg/genericapiserver/storage_factory.go

```diff
@@ -88,6 +88,11 @@ type groupResourceOverrides struct {
 	// of exposing one set of concepts.  autoscaling.HPA and extensions.HPA as a for instance
 	// The order of the slice matters!  It is the priority order of lookup for finding a storage location
 	cohabitatingResources []unversioned.GroupResource
+
+	// ignoreCohabitatingStorageVersion controls whether or not a cohabitating resource attempts to serialize into the same
+	// groupVersion as all other cohabitators.  If not, it uses the groupVersion assigned to itself instead of the one being
+	// used by the first enabled cohabitator.
+	ignoreCohabitatingStorageVersion bool
```

Comment:
> I'm really not sure this is a good idea. Does this mean if I write to /b the storage format becomes 'b' and then if I write the same object through /a it switches back to 'a'?

We are exploring options in https://github.com/openshift/origin/pull/9941, but the current thinking is that during our transition, we'd either disallow the skewed updates or wire them to work by running an internal to internal conversion and calling out to the alternate RESTStorage.  Neither option is desireable, but it allows steps to look like this:

1. cohabitation without full-fidelity conversion
2. full-fidelity conversions, but extensive enough drift we're worried version skewing servers storing in the same locations with non-native serializations.
3. full-fidelity conversions, with one serialization mechanism.
4. drop old endpoints.

We're at stage 3 with RC<->RS, we're at stage 1 or 2 with D<->DC.  I don't think either case is so special that it won't happen to us again.